data/tutorials/gs_01_a_first_hour_with_ocaml.md

```diff
@@ -1,47 +1,65 @@
+
+**Please note**: this tutorial assumes you have some programming experience and are familiar with things like functions, variables, loops, arrays, etc. 
+
+## A Brief Introduction to OCaml
+
+OCaml is a functional programming language, similar to Haskell, Rust, and Go. OCaml also supports imperative and object-oriented features, making it more versatile than many languages. OCaml enables you to construct code in a way that minimises bugs and maximises security. It’s a type-safe language, like Swift, meaning it won’t let you execute an operation of the same type, reducing errors and increasing readability. 
+
+OCaml is also fast! Once compiled, its speed is comparable to C, one of the fastest languages. OCaml 5.0 provides multicore support! This gives it the ability to utilise multiple cores available on modern processors, which speeds up execution of the program even further.
+
+Although most programmers associate OCaml with its applications in academia, it’s actually much more versatile. It’s currently used in the IT industry in companies such as Facebook, Bloomberg, Jane Street Capital, and Docker, to name just a few. In fact, all the exercises in the popular [Advent of Code](https://adventofcode.com/) site can be done in OCaml! It's a good place to practice once you know the basics. 
+
+For more detailed information on OCaml and reasons to include it in your programming repetoire, please read ["Why OCaml?"](https://www.ocaml.org/about).
+
 
 ## Running OCaml Programs
 
-To try small OCaml expressions, use an interactive toplevel, or REPL
-(Read-Eval-Print Loop). The `ocaml` command provides a basic toplevel
-(install `rlwrap` through your system package manager and run `rlwrap ocaml` instead to get history navigation).
+Since we'll be working with small OCaml expressions at first, using an interactive toplevel, or REPL
+(Read-Eval-Print Loop), will suffice. The `ocaml` command provides a basic toplevel, which we installed in [Up and Running](/docs/up-and-running). If you didn't go through that part or you find you don't have it installed, you can install it with `rlwrap` through your system package manager and run `rlwrap ocaml` instead to get history navigation.
 
-The alternative REPL [utop](https://github.com/ocaml-community/utop) may be installed
-through [opam](https://opam.ocaml.org/doc/Install.html) or your system package manager. It has
-the same basic interface but is much more convenient to use (history
+We also installed the alternative REPL [utop](https://github.com/ocaml-community/utop), which recommend for its simplicity, through [opam](https://opam.ocaml.org/doc/Install.html), but you can also install it through your system package manager. It has
+the same basic interface as `ocaml`, but is much more convenient to use (history
 navigation, auto-completion, etc).
```

Comment:
As a reader, this leaves me confused which of the two ways here I should prefer, and also I don't understand why I should install `rlwrap`, when I have already installed UTop during the previous tutorial.

To me, it looks like it would be healthier for our docs to deep-link into the Up and Running tutorial to tell ppl how to set up the REPL, instead of bringing up new ways of installing OCaml here.

```suggestion
We will be working with small OCaml expressions at first, thus, using an interactive toplevel, or REPL
(Read-Eval-Print Loop) will suffice.

We recommend to use UTop (see [here](/docs/up-and-running#using-the-ocaml-toplevel-with-utop)),
but alternatively, the basic toplevel of OCaml (`ocaml` command) can be used (for installation instructions,
see [Up and Running](/docs/up-and-running#installing-ocaml)).
```

Reply:
The `rlwrap` was mentioned in the original, so I was trying to give it more context/explanation here, but I think you're right.

Reply:
I can't commit this section for some reason, so I'll close this and change it manually.